pkg/scaling/scale_handler.go

```diff
@@ -0,0 +1,305 @@
+	case values.KindDeployments:
+		err := h.ScaleDeployment(namespacedName.Namespace, namespacedName.Name, 0)
+		if err != nil {
+			return fmt.Errorf("ScaleTargetToZero - Deployment: %w", err)
+		}
+	case values.KindRollout:
+		err := h.ScaleArgoRollout(namespacedName.Namespace, namespacedName.Name, 0)
+		if err != nil {
+			return fmt.Errorf("ScaleTargetToZero - Rollout: %w", err)
+		}
+	default:
+		return fmt.Errorf("unsupported target kind: %s", targetKind)
+	}
+	return nil
+}
+
+// ScaleDeployment scales the deployment to the provided replicas
+// TODO: use a generic logic to perform scaling similar to HPA/KEDA
+func (h *ScaleHandler) ScaleDeployment(ns, targetName string, replicas int32) error {
+	deploymentClient := h.kClient.AppsV1().Deployments(ns)
+	deploy, err := deploymentClient.Get(context.TODO(), targetName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("ScaleDeployment - GET: %w", err)
+	}
+
+	h.logger.Debug("Deployment found", zap.String("deployment", targetName), zap.Int32("current replicas", *deploy.Spec.Replicas), zap.Int32("desired replicas", replicas))
+	if deploy.Spec.Replicas != nil && *deploy.Spec.Replicas != replicas {
+		*deploy.Spec.Replicas = replicas
+		_, err = deploymentClient.Update(context.TODO(), deploy, metav1.UpdateOptions{})
+		if err != nil {
+			return fmt.Errorf("ScaleDeployment - Update: %w", err)
+		}
+		h.logger.Info("Deployment scaled", zap.String("deployment", targetName), zap.Int32("replicas", replicas))
+		return nil
+	}
+	h.logger.Info("Deployment already scaled", zap.String("deployment", targetName), zap.Int32("current replicas", *deploy.Spec.Replicas))
+	return nil
+}
+
+// ScaleArgoRollout scales the rollout to the provided replicas
+func (h *ScaleHandler) ScaleArgoRollout(ns, targetName string, replicas int32) error {
+	rollout, err := h.kDynamicClient.Resource(values.RolloutGVR).Namespace(ns).Get(context.TODO(), targetName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("ScaleArgoRollout - GET: %w", err)
+	}
+
+	currentReplicas := rollout.Object["spec"].(map[string]interface{})["replicas"].(int64)
+	h.logger.Info("Rollout found", zap.String("rollout", targetName), zap.Int64("current replicas", currentReplicas), zap.Int32("desired replicas", replicas))
+
+	if currentReplicas != int64(replicas) {
+		rollout.Object["spec"].(map[string]interface{})["replicas"] = replicas
+		_, err = h.kDynamicClient.Resource(values.RolloutGVR).Namespace(ns).Update(context.TODO(), rollout, metav1.UpdateOptions{})
+		if err != nil {
+			return fmt.Errorf("ScaleArgoRollout - Update: %w", err)
+		}
+		h.logger.Info("Rollout scaled", zap.String("rollout", targetName), zap.Int32("replicas", replicas))
+		return nil
+	}
+	h.logger.Info("Rollout already scaled", zap.String("rollout", targetName), zap.Int64("current replicas", currentReplicas))
+
+	return nil
+}
+
+func (h *ScaleHandler) UpdateKedaScaledObject(ctx context.Context, scaledObjectName, namespace string, paused bool) error {
```

Comment:
```suggestion
func (h *ScaleHandler) UpdateKedaScaledObjectPausedState(ctx context.Context, scaledObjectName, namespace string, paused bool) error {
```
Typically update functions should be touching all the fields relevant